It's me again :) http.compression more effective !

yvann opened the pull request:
Moved the "accept-encoding" part to be always effective, the compression will depend of the ES configuration, we only say that we support it.

ruflin commented:
@yvann Interesting. Can you please add at least one integration test with compression enabled so we can make sure this works as expected?

yvann commented:
Sure ! I just saw the last comment you did on my previous PR :)

ruflin commented:
@yvann Any updates here?

ruflin commented:
@yvann I would love to get this into master. Any updates here?

yvann commented:
I have no good excuse to not do it, let me the day to push something :) !

ruflin commented:
@yvann :+1:  :-D

yvann commented:
Rebased and added some tests, at least to show that the compression parameter doesn't affect the success of requests.
The behavior for the Http transport is :
- Elastica always says to ES it supports compression (via curl native support), to allow ES to send back a compressed response
- Depending of compression parameter, the request sent to ES is compressed or not

ruflin commented:
Merged, thx
